controller/canary_test.go

```diff
@@ -896,3 +898,52 @@ func TestCanaryRolloutStatusHPAStatusFields(t *testing.T) {
 	patch := f.getPatchedRollout(index)
 	assert.Equal(t, calculatePatch(r2, expectedPatchWithSub), patch)
 }
+
+func TestCanaryRolloutWithCanaryService(t *testing.T) {
+	f := newFixture(t)
+
+	canarySvc := newService("canary", 80, make(map[string]string))
```

Comment:
What is there a reason that you instantiate the selector to nothing? If we don't, that will allow us to also test the case where the selector is nil.

Reply:
Good point. Changed `make(map[string]string)` to nil.